tracker/tracker/src/main/modules/input.ts

```diff
@@ -131,60 +127,88 @@ export default function (app: App, opts: Partial<Options>): void {
   }
 
   const inputValues: Map<number, string> = new Map()
-  const checkableValues: Map<number, boolean> = new Map()
-  const registeredTargets: Set<number> = new Set()
+  const checkboxValues: Map<number, boolean> = new Map()
 
   app.attachStopCallback(() => {
     inputValues.clear()
-    checkableValues.clear()
-    registeredTargets.clear()
+    checkboxValues.clear()
   })
 
-  app.ticker.attach((): void => {
+  function trackInputValue(id: number, node: TextFeildElement) {
+    if (inputValues.get(id) === node.value) {
+      return
+    }
+    inputValues.set(id, node.value)
+    sendInputValue(id, node)
+  }
+
+  function trackCheckboxValue(id: number, value: boolean) {
+    if (checkboxValues.get(id) === value) {
+      return
+    }
+    checkboxValues.set(id, value)
+    app.send(SetInputChecked(id, value))
+  }
+
+  // The only way (to our knowladge) to track all kinds of input changes, including those made by JS
+  app.ticker.attach(() => {
     inputValues.forEach((value, id) => {
       const node = app.nodes.getNode(id) as HTMLInputElement
       if (!node) return inputValues.delete(id)
-      if (value !== node.value) {
-        inputValues.set(id, node.value)
-        if (!registeredTargets.has(id)) {
-          registeredTargets.add(id)
-          sendInputTarget(id, node)
-        }
-        sendInputValue(id, node)
-      }
+      trackInputValue(id, node)
     })
-    checkableValues.forEach((checked, id) => {
+    checkboxValues.forEach((checked, id) => {
       const node = app.nodes.getNode(id) as HTMLInputElement
-      if (!node) return checkableValues.delete(id)
-      if (checked !== node.checked) {
-        checkableValues.set(id, node.checked)
-        app.send(SetInputChecked(id, node.checked))
-      }
+      if (!node) return checkboxValues.delete(id)
+      trackCheckboxValue(id, node.checked)
     })
-  })
-  app.ticker.attach(Set.prototype.clear, 100, false, registeredTargets)
+  }, 5)
+
+  function sendInputChange(id: number, node: TextFeildElement, hesitationTime: number) {
+    trackInputValue(id, node)
+    const label = getInputLabel(node)
+    app.send(InputChange(id, label, hesitationTime))
+  }
 
   app.nodes.attachNodeCallback(
     app.safe((node: Node): void => {
       const id = app.nodes.getID(node)
       if (id === undefined) {
         return
       }
-      // TODO: support multiple select (?): use selectedOptions; Need send target?
+
+      // TODO: support multiple select (?): use selectedOptions;
       if (hasTag(node, 'select')) {
         sendInputValue(id, node)
-        app.attachEventListener(node, 'change', () => {
-          sendInputValue(id, node)
-        })
+        app.nodes.attachNodeListener(node, 'change', () => sendInputValue(id, node))
       }
-      if (isTextEditable(node)) {
-        inputValues.set(id, node.value)
-        sendInputValue(id, node)
+
```

Comment:
Here I fixed the code because it was easier than point out each small change in the review.